From 3b8a233e29918554f11fa63b2cc01b94a99e933b Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 16 May 2014 20:13:30 -0400
Subject: [md] dm-thin: use INIT_WORK_ONSTACK in noflush_work to avoid ODEBUG warning

Message-id: <1400271213-25480-7-git-send-email-snitzer@redhat.com>
Patchwork-id: 79876
O-Subject: [RHEL7.1 PATCH 6/9] dm thin: use INIT_WORK_ONSTACK in noflush_work to avoid ODEBUG warning
Bugzilla: 1098643
RH-Acked-by: Joe Thornber <thornber@redhat.com>
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>

BZ: 1098643

Upstream commit fbcde3d8b9c2d97704b8ca299e5266147b24c8ee
Author: Mike Snitzer <snitzer@redhat.com>
Date:   Tue Apr 29 11:22:04 2014 -0400

    dm thin: use INIT_WORK_ONSTACK in noflush_work to avoid ODEBUG warning

    Use INIT_WORK_ONSTACK to silence "ODEBUG: object is on stack, but not
    annotated".

    Reported-by: Zdenek Kabelac <zkabelac@redhat.com>
    Signed-off-by: Mike Snitzer <snitzer@redhat.com>
    Acked-by: Joe Thornber <ejt@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm-thin.c b/drivers/md/dm-thin.c
index 619d661..3259aff 100644
--- a/drivers/md/dm-thin.c
+++ b/drivers/md/dm-thin.c
@@ -1619,7 +1619,7 @@ static void noflush_work(struct thin_c *tc, void (*fn)(struct work_struct *))
 {
  struct noflush_work w;
 
- INIT_WORK(&w.worker, fn);
+ INIT_WORK_ONSTACK(&w.worker, fn);
  w.tc = tc;
  atomic_set(&w.complete, 0);
  init_waitqueue_head(&w.wait);
-- 
1.7.1